File: TeamCode/src/main/java/com/info1robotics/bobot/tasks/MoveTask.java
package com.info1robotics.bobot.tasks;

import com.qualcomm.robotcore.hardware.DcMotor;

public class MoveTask extends Task {
    private final double x;
    private final double y;
    private final double t;
    private final double power;
    private final int position;

    public MoveTask(double x, double y, double t, double power, int position) {
        this.x = x;
        this.y = y;
        this.t = t;
        this.power = power;
        this.position = position;
    }

    public boolean stillRunning() {
        return (Math.abs(context.mecanum.br.getCurrentPosition())>=position || Math.abs(context.mecanum.br.getCurrentPosition()) >= position
                || Math.abs(context.mecanum.br.getCurrentPosition())>=position || Math.abs(context.mecanum.br.getCurrentPosition()) >= position);
    }

    @Override
    public void tick() {
        if (stillRunning()) return;
        context.mecanum.br.setPower(0.0);
        context.mecanum.fr.setPower(0.0);
        context.mecanum.bl.setPower(0.0);
        context.mecanum.fl.setPower(0.0);
        state = State.FINISHED;
    }

    @Override
    public void run() {
        context.mecanum.br.setMode(DcMotor.RunMode.STOP_AND_RESET_ENCODER);
        context.mecanum.bl.setMode(DcMotor.RunMode.STOP_AND_RESET_ENCODER);
        context.mecanum.fr.setMode(DcMotor.RunMode.STOP_AND_RESET_ENCODER);
        context.mecanum.fl.setMode(DcMotor.RunMode.STOP_AND_RESET_ENCODER);
        context.mecanum.vectorMove(x, y, t, power);
    }
}

File: TeamCode/src/main/kotlin/com/info1robotics/bobot/tasks/MoveTask.kt
package com.info1robotics.bobot.tasks

import com.qualcomm.robotcore.hardware.DcMotor
import org.apache.commons.math3.analysis.function.Abs
import kotlin.math.abs
import kotlin.math.max

class MoveTask(val x: Double, val y: Double, val t: Double, val power: Double, val position:Int ):Task() {
    fun stillRunning(): Boolean {
       return (abs(context.mecanum.br.currentPosition)>=position||abs(context.mecanum.br.currentPosition)>=position
                || abs(context.mecanum.br.currentPosition)>=position||abs(context.mecanum.br.currentPosition)>=position)
    }
   override fun tick() {
        while (stillRunning()){}
            context.mecanum.br.power=.0;
            context.mecanum.fr.power=.0;
            context.mecanum.bl.power=.0;
            context.mecanum.fl.power=.0;
            state=State.FINISHED;

    }

    override fun run() {

        context.mecanum.bl.mode=DcMotor.RunMode.STOP_AND_RESET_ENCODER
        context.mecanum.fl.mode=DcMotor.RunMode.STOP_AND_RESET_ENCODER
        context.mecanum.fr.mode=DcMotor.RunMode.STOP_AND_RESET_ENCODER
        context.mecanum.br.mode=DcMotor.RunMode.STOP_AND_RESET_ENCODER
        context.mecanum.fl.mode=DcMotor.RunMode.RUN_USING_ENCODER;
        context.mecanum.fr.mode=DcMotor.RunMode.RUN_USING_ENCODER
        context.mecanum.bl.mode=DcMotor.RunMode.RUN_USING_ENCODER;
        context.mecanum.br.mode=DcMotor.RunMode.RUN_USING_ENCODER
        context.mecanum.vectorMove(x,y,t,power)
        println(context.mecanum.fl.power)
        println(context.mecanum.br.power)
        println(context.mecanum.fr.power)
        println(context.mecanum.bl.power)
    }
}